src/ProgressBar/ProgressBar.tsx

```diff
@@ -39,20 +40,37 @@ const ProgressContainer = styled.span<StyledProgressContainerProps>`
 
 export type ProgressBarProps = React.PropsWithChildren & {bg?: string} & StyledProgressContainerProps & ProgressProp
 
-export const ProgressBar = ({
-  progress,
-  bg = 'success.emphasis',
-  barSize = 'default',
-  children,
-  ...rest
-}: ProgressBarProps) => {
-  if (children && progress) {
-    throw new Error('You should pass `progress` or children, not both.')
-  }
-
-  return (
-    <ProgressContainer barSize={barSize} {...rest}>
-      {children ?? <Item progress={progress} sx={{backgroundColor: bg}} />}
-    </ProgressContainer>
-  )
-}
+export const ProgressBar = forwardRef<HTMLSpanElement, ProgressBarProps>(
+  ({progress, bg = 'success.emphasis', barSize = 'default', children, ...rest}: ProgressBarProps, forwardRef) => {
+    if (children && progress) {
+      throw new Error('You should pass `progress` or children, not both.')
+    }
+
+    if (
+      children &&
+      typeof (rest as React.AriaAttributes)['aria-valuenow'] === 'undefined' &&
+      typeof (rest as React.AriaAttributes)['aria-valuetext'] === 'undefined'
+    ) {
+      throw new Error(
+        'Use the `aria-valuenow` or `aria-valuetext` so screen reader users can determine the `progress`.',
+      )
+    }
+
+    const progressAsNumber = typeof progress === 'string' ? parseInt(progress) : progress
+
+    const ariaAttributes = {
+      'aria-valuenow': progressAsNumber ? Math.round(progressAsNumber) : undefined,
```

Comment:
Oh wait... I need it in the case that the value is a `number` not a string

Reply:
@siddharthkp one note with `parseInt`, I believe it will infer the base if no `radix` is provided based on: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#radix

> If 0 or not provided, the radix will be inferred based on string's value. Be careful — this does not always default to 10!

Reply:
Oh wow, good note!